stages/org.osbuild.ostree.coreos.aleph

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python3
+"""
+Create aleph version file for the deployment.
+"""
+
+
+import os
+import sys
+
+import osbuild.api
+from osbuild.util import ostree
+
+import json
+
+CAPABILITIES = ["CAP_MAC_ADMIN"]
+
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "properties": {
+    "coreos_compat": {
+      "description": "boolean to allow for CoreOS aleph version backwards compatibility",
+      "type": "boolean"
+    },
+    "deployment": {
+      "additionalProperties": false,
+      "required": ["osname", "ref"],
+      "properties": {
+        "osname": {
+          "description": "Name of the stateroot to be used in the deployment",
+          "type": "string"
+        },
+        "ref": {
+          "description": "OStree ref to create and use for deployment",
+          "type": "string"
+        },
+        "serial": {
+          "description": "The deployment serial (usually '0')",
+          "type": "number",
+          "default": 0
+        }
+      }
+    }
+  }
+}
+"""
+
+def write_to_file(tree, filename, data, coreos_compat):
+    with open(tree + "/" + filename, "w") as f:
+        json.dump(data, f, indent=4, sort_keys=True)
+        f.write("\n")
+    f.close()
+
+    # create a symlink for backwards compatibility with CoreOS
+    if coreos_compat:
+      cwd = os.getcwd()
+      os.chdir(tree)
+      os.symlink(filename, ".coreos-aleph-version.json")
+      os.chdir(cwd)
```

Comment:
is the chdir here required?

Reply:
When making symlinks, you can either link them through absolute paths or relative paths. Originally, I specified the absolute path for the symlink. However, since we build from an osbuild buildroot and then later copy it over to the root partition, the absolute path previously specified is no longer valid. Thus, by using `chdir`, we can specify a relative path for the symlink (ie. link the coreos aleph with the general aleph file in the same directory) and it will not be affected when we copy over the `sysroot`.

Reply:
Yep, definitely want the symlink to be relative, but we can still achieve that without changing directory:

```
[dustymabe@media ~]$ mkdir footmp
[dustymabe@media ~]$ touch footmp/file
[dustymabe@media ~]$ python3
Python 3.11.6 (main, Oct  3 2023, 00:00:00) [GCC 12.3.1 20230508 (Red Hat 12.3.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.symlink('file', 'footmp/symlinktofile')
>>> 
[dustymabe@media ~]$ ls -l footmp/
total 4
-rw-r--r--. 1 dustymabe dustymabe 0 Nov 20 16:55 file
lrwxrwxrwx. 1 dustymabe dustymabe 4 Nov 20 16:56 symlinktofile -> file
```

Reply:
so.. we should just be able to change the code to:

```
os.symlink(filename, os.path.join(tree, ".coreos-aleph-version.json"))
```

I think